Sources/QuickCheck/ArbitraryDecodable.swift

```diff
@@ -0,0 +1,218 @@
+import Foundation
+import LCG
+import Prelude
+
+private let genInt8 = choose(Int(Int8.min)..<Int(Int8.max)).map(Int8.init)
+private let genInt16 = choose(Int(Int16.min)..<Int(Int16.max)).map(Int16.init)
+private let genInt32 = genInt.map(Int32.init)
+private let genInt64 = genInt.map(Int64.init)
+private let genUInt = choose(0..<2_000_000).map(UInt.init)
+private let genUInt8 = choose(0..<Int(UInt8.max)).map(UInt8.init)
+private let genUInt16 = choose(0..<Int(UInt16.max)).map(UInt16.init)
+private let genUInt32 = choose(0..<2_000_000).map(UInt32.init)
+private let genUInt64 = choose(0..<2_000_000).map(UInt64.init)
+private let genFloat = genDouble.map(Float.init)
+
+private final class ArbitraryDecoder: Decoder {
+  var genState: GenState
+  var codingPath: [CodingKey] = []
+  let userInfo: [CodingUserInfoKey: Any] = [:]
+
+  init(seed: Seed, size: Int) {
+    self.genState = (seed, size)
+  }
+
+  func run<T>(_ gen: Gen<T>) -> T {
+    let (result, genState) = gen.run(self.genState)
+    self.genState = genState
+    return result
+  }
+
+  func container<Key>(keyedBy type: Key.Type) throws -> KeyedDecodingContainer<Key> where Key: CodingKey {
+    return .init(KeyedContainer(decoder: self))
+  }
+
+  func unkeyedContainer() throws -> UnkeyedDecodingContainer {
+    return UnkeyedContainer(decoder: self)
+  }
+
+  func singleValueContainer() throws -> SingleValueDecodingContainer {
+    return SingleValueContainer(decoder: self)
+  }
+
+  struct KeyedContainer<Key: CodingKey>: KeyedDecodingContainerProtocol {
+    let allKeys: [Key]
+    let codingPath: [CodingKey]
+    let decoder: ArbitraryDecoder
+
+    init(decoder: ArbitraryDecoder) {
+      let genKeys = arrayOf(genInt.map(Key.init(intValue:))) |> scale { $0 * 2 }
+
+      self.allKeys = decoder.run(genKeys).flatMap { $0 }
```

Comment:
Decoders don't decode `Dictionary` as a type, so I'm using the `arrayOf` generator to build up a random collection of keys.